src/Lti1p1Installation.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Packback\Lti1p3;
+
+class Lti1p1Installation
+{
+    private ?array $oauth_consumer_keys;
+    private ?array $oauth_consumer_secrets;
+
+    public static function new()
+    {
+        return new Lti1p1Installation();
+    }
+
+    public function compareOauthConsumerKeySign(array $lti1p3LaunchData, string $signature): bool
+    {
+        $clientId = is_array($lti1p3LaunchData['aud']) ? $lti1p3LaunchData['aud'][0] : $lti1p3LaunchData['aud'];
+        $issuerUrl = $lti1p3LaunchData['iss'];
+        $exp = $lti1p3LaunchData['exp'];
+        $nonce = $lti1p3LaunchData['nonce'];
+
+        $keys = $this->getOauthConsumerKeys();
+        $secrets = $this->getOauthConsumerSecrets();
+
+        for ($i = 0; $i < count($keys); $i++) {
```

Comment:
@erikdonohoo One final question (hopefully). In what circumstance would you have multiple potential keys to try to match against? I would assume that some combination of the `oauth_consumer_key`, `iss` and/or `aud` would be enough to definitively narrow it down to a single key when querying the database. Is that not the case?

Ideally I'd like to remove the `for` loop here, but I don't have enough context into how the 1.1 keys get queried.

Reply:
The short answer is key rotation. If you want to support rotating keys, you need to support multiple keys for a 1.1 installation. Having that be an array, enables that. Does that make sense @dbhynds ?

Reply:
Boo. Yeah, that makes sense. Alas, the `for` loop shall remain!